publication-doi-commons/src/main/java/no/unit/nva/publication/doi/PublicationMapper.java

```diff
@@ -91,23 +95,70 @@ public PublicationMapping fromDynamodbStreamRecord(DynamodbStreamRecord streamRe
      * @return Publication doi.Publication
      */
     public Publication fromDynamodbStreamRecordDao(DynamodbStreamRecordImageDao dao) {
+
         return Builder.newBuilder()
-            .withId(transformIdentifierToId(namespacePublication, dao.getIdentifier()))
-            .withInstitutionOwner(URI.create(dao.getPublisherId()))
+            .withId(transformIdentifierToId(namespacePublication, dao))
```

Comment:
read the JsonPointers and if the field exists add the value, else put null. Validation will not happen here.